app/services/reporting/agency_and_sp_report.rb

```diff
@@ -0,0 +1,43 @@
+module Reporting
+  class AgencyAndSpReport
+    attr_reader :report_date
+
+    def initialize(report_date = Time.zone.today)
+      @report_date = report_date
+    end
+
+    def agency_and_sp_report
+      idv_agency_ids = idv_sps.collect { |sp| sp.agency_id }.uniq
+
+      # We define auth as "not IDV", can include ial: 1 and ial: nil
+      auth_sp_count = ServiceProvider.active.count - idv_sps.count
+      auth_agency_count = Agency.count - idv_agency_ids.count
```

Comment:
for consistency, these DB queries should also match the same time bounding right?
```suggestion
      auth_sp_count = ServiceProvider.where('created_at < ?', report_date).active.count - idv_sps.count
      auth_agency_count = Agency.where('created_at < ?', report_date).count - idv_agency_ids.count
```

Reply:
I feel like another way we could do these would be with `Enumerable#partition` like:

```ruby
idv_sps, auth_sps = ServiceProvider.where('created_at < ?', report_date).active.partition { |sp| sp.ial >= 2 }
idv_agency_ids = idv_sps.map(&:agency_id).uniq
idv_agencies, auth_agencies = Agency.where('created_at < ?', report_date).partition { |ag| idv_agency_ids.include?(ag.id) }
```

Reply:
D'oh! Good catch with `created_at` on agencies.

I don't think I have ever used `partition` before, but it looks like precisely what the doctor ordered here!

Reply:
ok so we'll need to join agencies to SPs, so an agency will be active if it has an SP that is active